modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/jaxrs/extension/ObjectRelationshipEndpointsExtensionImpl.java

```diff
@@ -0,0 +1,153 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.object.rest.internal.jaxrs.extension;
+
+import com.liferay.object.related.models.ObjectRelationshipEndpointsExtension;
+import com.liferay.object.system.SystemObjectDefinitionMetadata;
+import com.liferay.osgi.service.tracker.collections.list.ServiceTrackerList;
+import com.liferay.osgi.service.tracker.collections.list.ServiceTrackerListFactory;
+import com.liferay.portal.kernel.util.HashMapDictionaryBuilder;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import javax.ws.rs.core.Application;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceReference;
+import org.osgi.service.component.ComponentFactory;
+import org.osgi.service.component.ComponentInstance;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.util.tracker.ServiceTrackerCustomizer;
+
+/**
+ * @author Luis Miguel Barcos
+ */
+@Component(service = ObjectRelationshipEndpointsExtension.class)
+public class ObjectRelationshipEndpointsExtensionImpl
+	implements ObjectRelationshipEndpointsExtension {
+
+	@Override
+	public void addSystemObjectRelationshipsEndpoints(
+		SystemObjectDefinitionMetadata systemObjectDefinitionMetadata) {
+
+		Set<String> basePaths = _applicationNames.keySet();
+
+		Stream<String> basePathsStream = basePaths.stream();
+
+		String systemObjectBasePath = basePathsStream.filter(
+			s -> {
+				String restContextPath =
+					systemObjectDefinitionMetadata.getRESTContextPath();
+
+				String basePath = "/" + restContextPath.split("/")[0];
+
+				return s.contains(basePath);
+			}
+		).findFirst(
+		).orElse(
+			""
+		);
+
+		String jaxrsApplicationName = _applicationNames.get(
```

Comment:
Not sure what we are trying to achieve here...

Maybe we can review it on a call? Or if you think it is not necessary I trust you with this